src/evict/evict_lru.c

```diff
@@ -1717,6 +1717,15 @@ __evict_walk_tree(WT_SESSION_IMPL *session, WT_EVICT_QUEUE *queue, u_int max_ent
     if (target_pages > remaining_slots)
         target_pages = remaining_slots;
 
+    /*
+     * Skip btrees other than history store if the cache pressure is high and HS content dominates
+     * cache. Evicting unclean non-HS pages can generate even more HS content and will not help with
+     * the cache pressure and will probably just amplify it further.
+     */
+    if (!WT_IS_HS(btree) && __wt_cache_almost_full(session) &&
+      __wt_cache_bytes_plus_overhead(cache, cache->bytes_hs) >= conn->cache_size / 4)
```

Comment:
I'm concerned about this for a few reasons. I'm not sure that 1/4 of the cache is a good metric.

I suspect this is working because the size you chose (25% of cache) is a little bit above the default dirty target, and in your workload the history store content is nearly all dirty. Given that it's the relationship to dirty data that is important, I think it would be better to tie the heuristic here to dirty cache usage directly.

Reply:
I need to look into if we track dirty cache content for HS specifically - but yes it does make sense to base decision on dirty data only rather than overcall HS cache usage.